### ISSUE TITLE
Fix deprecations

### ISSUE THREAD

**Issue by mhalbritter** [spring-io/initializr#1515]
- [ ] `io.spring.initializr.generator.project.ProjectGenerator#getCandidateProjectGenerationConfigurations`
- [ ] `io.spring.initializr.versionresolver.DefaultMavenVersionResolver`

**Comment by fulln** [comment]
can i accept this task?

**Comment by mhalbritter** [comment]
I don't think we can get rid of `ServiceLocator` in `io.spring.initializr.versionresolver.DefaultMavenVersionResolver`/

**Comment by Safkatul-Islam** [comment]
Hi! Hope everything's well with you.

I’d like to work on this issue. I can take care of the remaining deprecations if that’s okay.

**Comment by sri1873** [comment]
Hey if no one is working on this. I'd like to work on this issue

**Comment by mhalbritter** [comment]
Thanks for the offer! Go ahead!

**Comment by sri1873** [comment]
> when looking at the usages of getProjectGenerationConfigurationFactoryNames, this could be optimized, too.
> 
> in getCandidateProjectGenerationConfigurations, Class<?> type = resolveClass(candidate); is used to instantiate the class. This could be avoided if we don't return Strings from getProjectGenerationConfigurationFactoryNames but the class directly.](https://github.com/spring-io/initializr/pull/1690#issuecomment-3222839046)
> 

Hey @mhalbritter, So far, I replaced `loadFactoryNames()` with `load()` similar to that of PR #1690 and tried to optimise `getProjectGenerationConfigurationFactoryNames` to return the class directly instead of a list of strings, but `ProjectGenerationConfiguration` is an annotation used as the key in `spring.factories`, not a factory interface. The discovered classes are annotated with `@ProjectGenerationConfiguration` rather than assignable to it, so `load(...)` fails during runtime with _"not assignable to factory type"_. Do you have a preferred replacement for this annotation-based lookup, or would you prefer keeping `loadFactoryNames()` (with suppression) until a suitable alternative is available?

**Comment by mhalbritter** [comment]
Ah, i haven't thought of that. 

> or would you prefer keeping loadFactoryNames() (with suppression) until a suitable alternative is available?

I guess we need to keep it.
